### PR TITLE
encryption: Lazily decrypt `user_external_accounts`

### PR DESCRIPTION
**Pull request by efritz**
Stacked on #40282. This PR wraps the `user_external_accounts.auth_data` and `user_external_accounts.account_data` fields in an encryptable struct.

## Test plan

main-dry-run: https://buildkite.com/sourcegraph/sourcegraph/builds/166839

### CODE REVIEW

File: internal/extsvc/testing.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package extsvc
+
+import "github.com/google/go-cmp/cmp"
+
+var AccountComparer = cmp.Comparer(func(a, b Account) bool {
+	if false ||
```
Comment (eseliger) [inline internal/extsvc/testing.go:6]: is this a readability thing or something I don't understand?

Comment (efritz) [reply internal/extsvc/testing.go:6]: Was just to have all conditions aligned (I agree it looks dumb)

**Review by unknwon — APPROVED** [review]

**Comment by unknwon** [comment]
Nice!

**Review by vrto — APPROVED** [review]
LGTM, just one small qq

File: internal/database/external_accounts.go [hunk 2]
```diff
@@ -408,38 +393,23 @@ func (s *userExternalAccountsStore) ListBySQL(ctx context.Context, querySuffix *
...
 		if authData.Valid {
-			decryptedAuthData, err := s.maybeDecrypt(ctx, authData.String, keyID)
-			if err != nil {
-				return nil, err
-			}
-			if decryptedAuthData != "" {
-				jAuthData := json.RawMessage(decryptedAuthData)
-				acct.AuthData = &jAuthData
-			}
+			acct.AuthData = extsvc.NewEncryptedData(authData.String, keyID, s.getEncryptionKey())
```
[the remark this answers is not in the document]
Comment (efritz) [reply internal/database/external_accounts.go:409]: The underlying `AuthData` is a `JSONEncryptable`, which does the json encoding prior to encryption and json decoding after decryption.

[the remark this answers is not in the document]
Comment (vrto) [reply internal/database/external_accounts.go:409]: Thanks for the explanation!

**Pull request merged**